### PR TITLE
Change build to use yarn instead of npm 🧶

### PR DESCRIPTION
**Pull request by natemcmaster**
Resolves https://github.com/aspnet/AspNetCore/issues/7563 

This PR has two commits. I definitely recommend reviewing the first commit separately because the result of running `yarn import` (converts package-lock.json to yarn.lock) changed like a bajillion lines of code.

### CODE REVIEW

**Review by analogrelay — COMMENTED** [review]
Looks OK to me but I don't have the context of all the build system changes that have happened over the past couple of months to feel confident signing off completely :)

**Review by ryanbrandenburg — APPROVED** [review]
I'm not super familiar with yarn but this looks good.

**Comment by ryanbrandenburg** [comment]
Looks like you need to regenerate your ref files and fix a reference to Yarn.msbuild?

File: src/SignalR/clients/ts/signalr/README.md [hunk 1]
```diff
@@ -4,6 +4,7 @@ JavaScript and TypeScript clients for SignalR for ASP.NET Core
 
 ```bash
 npm install @aspnet/signalr
+# or "yarn add @aspnet/signalr"
```
Comment (BrennanConroy) [inline src/SignalR/clients/ts/signalr/README.md]: Whats with using `#` here?

File: src/SignalR/clients/ts/signalr-protocol-msgpack/README.md [hunk 2]
```diff
@@ -4,6 +4,7 @@ MsgPack support for SignalR for ASP.NET Core
 
 ```bash
 npm install @aspnet/signalr-protocol-msgpack
+# or "yarn add @aspnet/signalr-protocol-msgpack"
```
Comment (BrennanConroy) [inline src/SignalR/clients/ts/signalr-protocol-msgpack/README.md]: And here

**Review by BrennanConroy — APPROVED** [review]

**Review by dougbu — APPROVED** [review]
LGTM once builds succeed

To confirm: Builds will do the needful and `yarn` is not a prerequisite. Correct?

File: src/SignalR/clients/ts/signalr/README.md (hunk 1 above)
Comment (natemcmaster) [reply src/SignalR/clients/ts/signalr/README.md]: It's a code comment. Users should run either npm install or yarn add, but not both. Open to suggestions for making that more obvious

Comment (BrennanConroy) [reply src/SignalR/clients/ts/signalr/README.md]:
```bash
npm install @aspnet/signalr
```
or
```bash
yarn add @aspnet/signalr
```

Could work

**Review by SteveSandersonMS — APPROVED** [review]
https://github.com/natemcmaster/Yarn.MSBuild says that it installs Yarn for you so you don't need to install it globally. Does that mean it always prefers the local installation even if you have a global one?

If so, great! It would definitely be preferable for the repo to control the version of Yarn we're using, so we don't get any inconsistencies based on which versions people have installed globally.

If it ever does use the globally-installed version, maybe that's something we could look at changing if it turns out to cause problems.

**Comment by natemcmaster** [comment]
> Does that mean it always prefers the local installation even if you have a global one?

Yes, by default Yarn.MSBuild uses a local version of yarn. Yarn.MSBuild 1.12.3 will use yarn 1.12.3.

That said, I've just noticed the latest stable yarn is 1.13.0. I'll update Yarn.MSBuild and update this PR to use the latest verison of yarn.

**Comment by natemcmaster** [comment]
> @dougbu: To confirm: Builds will do the needful and `yarn` is not a prerequisite. Correct?

Confirmed. Machine prerequisites have not changed. You still only need NodeJS. Yarn.MSBuild takes care of acquiring yarn. You only need to install yarn to your machine if you want to make changes to one of the npm packages. (`npm i -g yarn` is the easiest way to do that.)

File: src/SignalR/clients/ts/signalr/README.md (hunk 1 above)
Comment (natemcmaster) [reply src/SignalR/clients/ts/signalr/README.md]: ok, updated the readme

**Pull request merged**